seumich/views.py

```diff
@@ -179,7 +179,9 @@ class StudentView(LoginRequiredMixin, UserLogPageViewMixin, TemplateView):
 
     def get_context_data(self, student, **kwargs):
         context = super(StudentView, self).get_context_data(**kwargs)
-        selected_student = get_object_or_404(Student, username=student)
+        selected_student = get_object_or_404(
+            Student.objects.prefetch_related('studentclasssitestatus_set__status'), 
+            username=student)
```

Comment:
Did we gain some performance here? It looks like we're using `prefetch_related` again with `status` later on, so I wonder if maybe the later queries could be simplified. The `many-to-many` relationship makes this somewhat complex, but perhaps there's a way to `prefetch_related` everything here and then just access everything through related objects?

Reply:
Yeah, this is actually being gained in the student_class_site_detail  template when status is being called. By prefetch_related on the status here it doesn't need to query it on every iteration in the for loop there! 

`{% with status=element.status %}`